### PR TITLE
Update the testsuite to use the new mink/driver-testsuite

### PR DESCRIPTION
**Pull request by stof**
Hi,

We are in the process of extracting the Mink driver testsuite to a dedicated package rather than having it inside Mink itself, to simplify the maintenance. So I updated your driver to use this setup as well to allow us to complete the migration (the last step will be the removal of the driver testsuite from the Mink repo).

### CODE REVIEW

File: bin/run-tests.sh [hunk 1]
```diff
@@ -18,22 +18,16 @@ stop_services(){
...
-  else
-    php -S 127.0.0.1:6789 2>&1 >> /dev/null &
-  fi
+  ${CURRENT_DIR}/bin/mink-test-server > /dev/null 2>&1 &
   sleep 2
 }
 
 mkdir -p /tmp/jcalderonzumba/phantomjs
-stop_services
+stop_services || true
```
Comment (stof) [inline bin/run-tests.sh]: this avoids making the script fail when services are not running yet.

File: composer.json [hunk 2]
```diff
@@ -26,11 +26,8 @@
     "jcalderonzumba/gastonjs": "~1.0"
   },
   "require-dev": {
-    "symfony/process": "~2.3",
-    "symfony/phpunit-bridge": "~2.7",
-    "symfony/css-selector": "~2.1",
-    "phpunit/phpunit": "~4.6",
-    "silex/silex": "~1.2"
```
Comment (stof) [inline composer.json]: Silex and Process are left-over from the split of gastonjs (which is the one needing them for its tests). CssSelector is a Mink dependency (and your requirement was forbidding to use latest version of Symfony).
And the PHPUnit Bridge is enforced by the new mink/driver-testsuite repo (and also allowing newer versions of the package)

File: integration_tests.xml [hunk 3]
```diff
@@ -1,38 +1,37 @@
...
-        <server name="WEB_FIXTURES_HOST" value="http://127.0.0.1:6789"/>
         <!-- where driver will connect to -->
         <server name="DRIVER_URL" value="http://127.0.0.1:8510/"/>
         <server name="TEMPLATE_CACHE_DIR" value="/tmp/jcalderonzumba/phantomjs"/>
     </php>
 
     <filter>
         <whitelist>
-            <directory>./src/Behat/Mink/Driver</directory>
+            <directory>./src/</directory>
```
Comment (stof) [inline integration_tests.xml]: this whitelist is to make it possible to collect code coverage. The previous rule was not matching anything, which broke the feature

File: integration_tests.xml [hunk 4]
```diff
@@ -1,38 +1,37 @@
...
+            <!--<directory>vendor/mink/driver-testsuite/tests/Css</directory>-->
+            <!--<file>vendor/mink/driver-testsuite/tests/Basic/StatusCodeTest.php</file>-->
+            <!--<file>vendor/mink/driver-testsuite/tests/Basic/HeaderTest.php</file>-->
+            <!--<file>vendor/mink/driver-testsuite/tests/Basic/NavigationTest.php</file>-->
         </testsuite>
     </testsuites>
 
     <php>
         <var name="driver_config_factory" value="Behat\Mink\Tests\Driver\PhantomJSConfig::getInstance"/>
-        <server name="WEB_FIXTURES_HOST" value="http://127.0.0.1:6789"/>
```
Comment (stof) [inline integration_tests.xml]: the new mink/driver-testsuite now makes this variable optional, defaulting to localhost:8002 which is what is used by the script shipped in it. So it is easier to omit it.

**Comment by stof** [comment]
The test failure also happens in master. It is not related to this PR
And we have the same failure on Selenium2Driver. So we will investigate on the Mink side.

**Comment by stof** [comment]
Btw, I found 2 weird things in your setup:

- why stopping the testsuite on the first failure on Travis ? This prevents seeing the full status of the driver
- why restarting gastonjs at the end of run-test.sh ? This makes it stay running after running tests

**Comment by jcalderonzumba** [comment]
@stof Very good points

1) Probably i was just testing travis and wanting to stop as soon an error was catched and it stayed like that forever.

2) That is really weird, however i think i left that on purpose for my local development environment testing. For travis that last line "start_browser_api" can be removed to be honest.

**Comment by stof** [comment]
> For travis that last line "start_browser_api" can be removed to be honest.

yeah, it would also remove the need to stop the browser api again in .travis.yml

**Comment by jcalderonzumba** [comment]
Yes, that line also deals with local manual testing, because in local i am constantly executing run-tests.sh so i needed that to make sure phantomjs was "freshly" started.

**Comment by jcalderonzumba** [comment]
@stof I just merged this PR and pushed a new tag for the driver but i realised now, will this introduce backwards incompatible changes due to setSession function changes? (i bumped from 0.3.0 to 0.3.1)

**Comment by stof** [comment]
@jcalderonzumba no it will not. ``setSession`` still exists (it is defined by CoreDriver itself)

**Pull request merged**